pandas/tests/io/excel/test_readers.py

```diff
@@ -1275,12 +1247,6 @@ def test_read_excel_squeeze(self, read_ext):
             expected = Series([1, 2, 3], name="a")
             tm.assert_series_equal(actual, expected)
 
-    def test_deprecated_kwargs(self, read_ext):
-        with tm.assert_produces_warning(FutureWarning, raise_on_extra_warnings=False):
```

Comment:
Could you turn this test into checking that non-keyword arguments here raise?

Reply:
Nice idea - wondering if there's a comprehensive way we should be testing for that (maybe part of test_api) as we do this with more functions

Reply:
Done

Reply:
> Nice idea - wondering if there's a comprehensive way we should be testing for that (maybe part of test_api) as we do this with more functions

Yeah it would be cool to have a test that nails down signature definitions and even alignment between DataFrame/Series methods